apidef/oas/middleware.go

```diff
@@ -429,3 +432,76 @@ func (a *Allowance) ExtractTo(endpointMeta *apidef.EndPointMeta) {
 	endpointMeta.Disabled = !a.Enabled
 	endpointMeta.IgnoreCase = a.IgnoreCase
 }
+
+func (ps Paths) fillCache(ep apidef.ExtendedPathsSet) {
+	advanceCacheConfig := &ep.AdvanceCacheConfig
+
+	for _, advCache := range *advanceCacheConfig {
+		if ps[advCache.Path] != nil {
+			plugins := ps[advCache.Path].getMethod(advCache.Method)
+			plugins.fillCache(advCache)
+
+		}
+	}
+}
+
+func (p *Plugins) fillCache(cacheMeta apidef.CacheMeta) {
+	if p.Cache == nil {
+		p.Cache = &CacheMiddleware{}
+	}
+	p.Cache.Fill(cacheMeta)
+}
+
+func (p *Plugins) extractCacheTo(ep *apidef.ExtendedPathsSet, path string, method string) {
+	advanceCacheConfig := ep.AdvanceCacheConfig
+	if advanceCacheConfig == nil {
+		advanceCacheConfig = []apidef.CacheMeta{}
+		newCache := apidef.CacheMeta{}
+		newCache.CacheOnlyResponseCodes = p.Cache.CacheResponseCodes
+		newCache.Disabled = !p.Cache.Enabled
+		newCache.CacheKeyRegex = p.Cache.CacheByRegex
+		newCache.Method = method
+		newCache.Path = path
+
+		advanceCacheConfig = append(advanceCacheConfig, newCache)
+		ep.AdvanceCacheConfig = advanceCacheConfig
+		return
+	}
+
+	founded := false
+	for _, advCache := range advanceCacheConfig {
+		if path == advCache.Path && method == advCache.Method && !founded {
+			founded = true
+			break
+		}
+	}
+	if !founded {
+		newCache := apidef.CacheMeta{}
+		newCache.CacheOnlyResponseCodes = p.Cache.CacheResponseCodes
+		newCache.Disabled = !p.Cache.Enabled
+		newCache.CacheKeyRegex = p.Cache.CacheByRegex
+		newCache.Method = method
+		newCache.Path = path
+
+		advanceCacheConfig = append(advanceCacheConfig, newCache)
+		ep.AdvanceCacheConfig = advanceCacheConfig
+	}
+}
+
+type CacheMiddleware struct {
+	Enabled            bool   `bson:"enabled" json:"enabled"`
+	CacheByRegex       string `bson:"cacheByRegex,omitempty" json:"cacheByRegex,omitempty"`
+	CacheResponseCodes []int  `bson:"cacheResponseCodes,omitempty" json:"cacheResponseCodes,omitempty"`
+}
+
+func (a *CacheMiddleware) Fill(cm apidef.CacheMeta) {
+	a.Enabled = !cm.Disabled
+	a.CacheByRegex = cm.CacheKeyRegex
+	a.CacheResponseCodes = cm.CacheOnlyResponseCodes
+}
+
+func (a *CacheMiddleware) ExtractTo(endpointMeta *apidef.CacheMeta) {
```

Comment:
where is this called?

Reply:
This is not called but I made it due to similar semantic we do everywhere